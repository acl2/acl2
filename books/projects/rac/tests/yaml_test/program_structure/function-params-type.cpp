// RAC begin

struct A { int a; };
struct B { int a; };

int foo(int a, A b) { return a; }


int bar() {
  B s;
  return foo(s, 0);
}

// RAC end
