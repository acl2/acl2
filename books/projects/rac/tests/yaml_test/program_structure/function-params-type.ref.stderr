function-params-type.cpp:11 (13-14): Expected `int`, got `struct {int a;}`
 11 |   return foo(s, 0);
    |              ^
note: Defined here: 
  6 | int foo(int a, A b) { return a; }
    | ^^^^^^^^^^^^^^^^^^^

function-params-type.cpp:11 (16-17): Expected `A`, got `int`
 11 |   return foo(s, 0);
    |                 ^
note: Defined here: 
  6 | int foo(int a, A b) { return a; }
    | ^^^^^^^^^^^^^^^^^^^
